api/app/schemas/hfi_calc.py

```diff
@@ -36,11 +36,95 @@ class StationDaily(BaseModel):
 
 
+required_daily_fields = ['temperature',
+                         'relative_humidity',
+                         'wind_speed',
+                         'wind_direction',
+                         'precipitation',
+                         'intensity_group']
+
+
 class StationDailyResponse(BaseModel):
     """ Response that includes list of station daily data."""
     dailies: List[StationDaily]
 
 
+class ValidatedStationDaily(BaseModel):
+    """
+    Station daily with validity flag
+    """
+    daily: StationDaily
+    valid: bool
+
+
+class FireStartRange(BaseModel):
+    """
+    User facing label, value and lookup table of fire starts to prep level
+    """
+    label: str
+    value: int
+    lookup_table: Mapping[int, int]
+
+
+lowest_fire_starts = FireStartRange(label='0-1', value=1, lookup_table={1: 1, 2: 1, 3: 2, 4: 3, 5: 4})
+one_2_two_starts = FireStartRange(label='1-2', value=2, lookup_table={1: 1, 2: 2, 3: 3, 4: 4, 5: 5})
+two_2_three_starts = FireStartRange(label='2-3', value=3, lookup_table={1: 2, 2: 3, 3: 4, 4: 5, 5: 6})
+three_2_six_starts = FireStartRange(label='3-6', value=6, lookup_table={1: 3, 2: 4, 3: 5, 4: 6, 5: 6})
+highest_fire_starts = FireStartRange(label='6+', value=7, lookup_table={1: 4, 2: 5, 3: 6, 4: 6, 5: 6})
+all_ranges = [lowest_fire_starts, one_2_two_starts,
+              two_2_three_starts, three_2_six_starts, highest_fire_starts]
+
+
+class DailyResult(BaseModel):
+    """
+    Prep level, MIG, fire starts and station daily results for a day in a prep week
+    """
+    dateISO: str
+    dailies: List[ValidatedStationDaily]
+    fire_starts: FireStartRange
+    mean_intensity_group: Optional[float]
+    prep_level: Optional[float]
+
+
+class PlanningAreaResult(BaseModel):
+    """
+    Mean prep level / max intensity group,
+    dailies and validity status of station dailies in a planning area
+    """
+    all_dailies_valid: bool
+    highest_daily_intensity_group: Optional[float]
+    mean_prep_level: Optional[float]
+    daily_results: List[DailyResult]
+
+
+class HFIResultResponse(BaseModel):
+    """
+    Response that contains daily data, num prep days, selected station codes,
+    selected fire centre, fire starts, HFI results
+    """
+    num_prep_days: int
+    selected_prep_date: datetime
+    start_time_stamp: Optional[int]
+    end_time_stamp: Optional[int]
+    selected_station_codes: List[int]
+    selected_fire_center: Optional[str]
+    planning_area_hfi_results: Mapping[str, PlanningAreaResult]
+    planning_area_fire_starts: Mapping[str, List[FireStartRange]]
+
+
+class HFIResultRequest(BaseModel):
+    """
+    Request that contains inputs necessary for calculating HFI
+    """
+    num_prep_days: int
+    selected_prep_date: datetime
+    start_time_stamp: Optional[int]
+    end_time_stamp: Optional[int]
+    selected_station_codes: List[int]
+    selected_fire_center: Optional[str]
+    planning_area_fire_starts: Mapping[str, List[FireStartRange]]
```

Comment:
Another issue here - I'm guessing the first key is the planning area, but you also need a key for the day. Each day in the prep period.